app/frontend/modules/filter_toggle.js

```diff
@@ -0,0 +1,112 @@
+export class FilterToggle {
+  constructor (options) {
+    this.options = options
+    this.container = this.options.toggleButton.container
+  }
+
+  setupResponsiveChecks () {
+    this.mq = window.matchMedia(this.options.bigModeMediaQuery)
+    this.mq.addListener(this.checkMode.bind(this))
+    this.checkMode(this.mq)
+  }
+
+  checkMode (mq) {
+    if (mq.matches) {
+      this.enableBigMode()
+    } else {
+      this.enableSmallMode()
+    }
+  }
+
+  enableBigMode () {
+    this.showMenu()
+    this.removeMenuButton()
+    this.removeCloseButton()
+  }
+
+  enableSmallMode () {
+    this.options.filter.container.setAttribute("tabindex", "-1")
+    this.hideMenu()
+    this.addMenuButton()
+    this.addCloseButton()
+  }
+
+  addCloseButton () {
+    if (this.options.closeButton) {
+      this.closeButton = document.createElement("button")
+      this.closeButton.classList.add("app-filter__close")
+      this.closeButton.innerText = this.options.closeButton.text
+      this.closeButton.type = 'button'
+      this.closeButton.addEventListener('click', this.onCloseClick.bind(this))
+
+      this.options.closeButton.container.append(this.closeButton)
+    }
+  }
+
+  onCloseClick () {
+    this.hideMenu()
+    this.menuButton.focus()
+  }
+
+  removeCloseButton () {
+    if (this.closeButton) {
+      this.closeButton.remove()
+      this.closeButton = null
+    }
+  }
+
+  addMenuButton () {
+    this.menuButton = document.createElement("button")
+    this.menuButton.setAttribute("aria-expanded", "false")
+    this.menuButton.setAttribute("aria-has-popup", "true")
+    this.menuButton.classList.add("govuk-button", this.options.toggleButton.classes, "app-filter-toggle__button")
+    this.menuButton.innerText = this.options.toggleButton.showText
+    this.menuButton.type = "button"
+    this.menuButton.addEventListener("click", this.onMenuButtonClick.bind(this))
+
+    this.options.toggleButton.container.prepend(this.menuButton)
+  }
+
+  removeMenuButton () {
+    if (this.menuButton) {
+      this.menuButton.remove()
+      this.menuButton = null
+    }
+  }
+
+  hideMenu () {
+    if (this.menuButton) {
+      this.menuButton.setAttribute("aria-expanded", "false")
+      this.menuButton.innerText = this.options.toggleButton.showText
+    }
+    this.options.filter.container.setAttribute("hidden", true)
+  }
+
+  showMenu () {
+    if (this.menuButton) {
+      this.menuButton.setAttribute("aria-expanded", "true")
+      this.menuButton.innerText = this.options.toggleButton.hideText
+    }
+    this.options.filter.container.removeAttribute("hidden")
+  }
+
+  onMenuButtonClick () {
+    this.toggle()
+  }
+
+  toggle () {
+    if (this.menuButton.getAttribute("aria-expanded") === "false") {
```

Comment:
shouldn't `aria-expanded` only be used for accessibility? Maybe using `display` or `hidden` attributes would be better?

Reply:
This attribute is used to inform the user the state of the element this button controls, so it’s nothing to do with visibility (when the filters are hidden, the `hidden` attribute is applied to the `app-filter-layout__filter` container.)

Reply:
Do you mean check the value of the `hidden` attribute, as opposed to the `aria-expanded` attribute?

The good thing about using an `aria` attribute (as opposed to a class) is it means it has to be accessible for it to work; there’s a direct link between what is being described and what is happening.